File: Check File SHA1 Hash/check_sha1.py
#!/usr/bin/env python

import os

#Define Formated Text Colors
class bcolors:
    HEADER = '\033[95m'
    OKBLUE = '\033[94m'
    OKGREEN = '\033[92m'
    WARNING = '\033[93m'
    FAIL = '\033[91m'
    ENDC = '\033[0m'
    BOLD = '\033[1m'
    UNDERLINE = '\033[4m'



#Get file to check from user
print bcolors.OKBLUE + "\nWhat's the original SHA1? " + bcolors.ENDC
sha1_to_check=raw_input()

print bcolors.OKBLUE + "\nWhat file do you want to check? " + bcolors.ENDC
print bcolors.WARNING + "* Drag file to terminal" + bcolors.ENDC
print bcolors.WARNING + "* Make sure there are no spaces in the file name\n" + bcolors.ENDC
filepath=raw_input()

downloaded_sha1 = (os.popen("/usr/bin/openssl sha1 %s | awk '{print $2}'" % filepath)).read().rstrip()


#sanity check
print bcolors.HEADER + "\n  Original SHA1: " + sha1_to_check + bcolors.ENDC
print bcolors.HEADER + "  Current  SHA1: " + downloaded_sha1 + "\n" + bcolors.ENDC

if downloaded_sha1 == sha1_to_check:
	print bcolors.OKGREEN + "  File's SHA1 Match\n" + bcolors.ENDC
elif downloaded_sha1 != sha1_to_check:
	print bcolors.FAIL + "  SHA1's DO NOT MATCH\n" + bcolors.ENDC
else:
	print bcolors.WARNING + "  Error: SHA1's could not be compared\n"

File: Check File SHA1 Hash/check_SHA1.rb
#!/usr/bin/ruby

#VARIABLES 
sha1_to_check="7a860d9d3f3512a9cdf770ca75c8fd2fdde4e0fc"


#define colored text
class String
	def red;		"\033[31m#{self}\033[0m" end
	def green;		"\033[32m#{self}\033[0m" end
	def brown;		"\033[33m#{self}\033[0m" end
	def blue;		"\033[34m#{self}\033[0m" end
	def magenta;	"\033[35m#{self}\033[0m" end
	def cyan;		"\033[36m#{self}\033[0m" end
	def gray;		"\033[37m#{self}\033[0m" end
	def yellow;		"\033[93m#{self}\033[0m" end
end

#Get file to check from user
puts "What file do you want to check? ".cyan
filepath=gets.chomp

downloaded_sha1=`/usr/bin/openssl sha1 #{filepath} | awk '{print $2}'`.chop

puts "  \n  Original SHA1: " + downloaded_sha1.yellow

if downloaded_sha1 == sha1_to_check
	puts "  Status: Files Match\n".green
else
	puts "  Status: Failed .... files do not match\n".red
end

 